src/main/java/cubicchunks/server/chunkio/async/forge/AsyncWorldIOExecutor.java

```diff
@@ -275,8 +280,10 @@ public static void tick() {
 		while (columnIter.hasNext()) {
 			AsyncColumnIOProvider task = columnIter.next();
 			if (task.isFinished()) {
-				if (task.hasCallbacks())
-					task.runSynchronousPart();
+				if (task.hasCallbacks()) {
+					task.runCallbacks();
+				}
+				task.runSynchronousPart();
```

Comment:
I don't know what I was thinking, either it's not needed or I will figure it out when fixing the above.